api/common/configentry_webhook.go

```diff
@@ -56,5 +64,25 @@ func ValidateConfigEntry(
 	if err := cfgEntry.Validate(enableConsulNamespaces); err != nil {
 		return admission.Errored(http.StatusBadRequest, err)
 	}
-	return admission.Allowed(fmt.Sprintf("valid %s request", cfgEntry.KubeKind()))
+	return admission.Patched(fmt.Sprintf("valid %s request", cfgEntry.KubeKind()), defaultingPatches...)
+}
+
+// defaultingPatches returns the patches needed to set fields to their
+// defaults.
+func defaultingPatches(cfgEntry ConfigEntryResource, enableConsulNamespaces bool, nsMirroring bool, consulDestinationNamespace string, nsMirroringPrefix string) ([]jsonpatch.Operation, error) {
```

Comment:
If we export this I think we can use it in serviceintentions and get rid of the duplicate code.

```go
	defaultingPatches, err := common.DefaultingPatches(&svcIntentions, v.EnableConsulNamespaces, v.EnableNSMirroring, v.ConsulDestinationNamespace, v.NSMirroringPrefix)
	if err != nil {
		return admission.Errored(http.StatusInternalServerError, err)
	}

```